src/main/java/com/algorand/algosdk/templates/PeriodicPayment.java

```diff
@@ -0,0 +1,111 @@
+
+    /**
+     * PeriodicPayment contract enables creating an account which allows the
+     * withdrawal of a fixed amount of assets every fixed number of rounds to a
+     * specific Algrorand Address. In addition, the contract allows to add
+     * timeout, after which the address can withdraw the rest of the assets.
+     *
+     * @param receiver address to receive the assets.
+     * @param amount amount of assets to transfer at every cycle.
+     * @param withdrawingWindow the number of blocks in which the user can withdraw the asset once the period start (must be < 1000).
+     * @param period how often the address can withdraw assets (in rounds).
+     * @param fee maximum fee per transaction.
+     * @param timeout a round in which the receiver can withdraw the rest of the funds after.
+     *
+     * @return PeriodicPayment contract.
+     */
+    public static ContractTemplate MakePeriodicPayment(final String receiver, final int amount, final int withdrawingWindow, final int period, final int fee, final int timeout) throws NoSuchAlgorithmException {
+        return MakePeriodicPayment(receiver, amount, withdrawingWindow, period, fee, timeout, null);
+    }
+
+    /**
+     * Need a way to specify the lease for testing.
+     */
+    protected static ContractTemplate MakePeriodicPayment(final String receiver, final int amount, final int withdrawingWindow, final int period, final int fee, final int timeout, final byte[] lease) throws NoSuchAlgorithmException {
+        if (!Lease.valid(lease)) {
+            throw new IllegalArgumentException("The lease should be an empty array or a 32 byte array.");
+        }
+
+        if (withdrawingWindow < 0 || withdrawingWindow > 1000) {
+            throw new IllegalArgumentException("The withdrawingWindow must be a positive number less than 1000");
+        }
+
+        ParameterValue[] values = new ParameterValue[] {
+                new IntParameterValue(fee),
+                new IntParameterValue(period),
+                new IntParameterValue(withdrawingWindow),
+                new IntParameterValue(amount),
+                new IntParameterValue(timeout),
+                new Base64ParameterValue(lease == null ? Lease.makeRandomLease() : lease),
+                new AddressParameterValue(receiver)
+        };
+
+        return ContractTemplate.inject(Encoder.decodeFromBase64(referenceProgram), referenceOffsets, values);
+    }
+
+    /**
+     * Return the withdrawal transaction to be sent to the network.
+     *
+     * @param contract contract containing information, this should be provided by the payer.
+     * @param firstValid first round the transaction should be valid.
+     * @param genesisHash genesis hash in base64.
+     * @return Signed withdrawal transaction.
+     */
+    public static SignedTransaction MakeWithdrawalTransaction(final byte[] contract, final int firstValid, final Digest genesisHash) throws IOException, NoSuchAlgorithmException {
```

Comment:
Why take the contract as byte[]? 
It will be nicer to have it in more specific data type, which is ContractTemplate.

Reply:
Good idea, I'll also need to enhance `ContractTemplate` to make it easier to initialize but that will be another good change.